src/FileHelper.php

```diff
@@ -464,15 +467,33 @@ public static function filterPath(string $path, array $options): bool
     {
         $path = str_replace('\\', '/', $path);
 
+        if (isset($options['filter'])) {
+            $result = call_user_func($options['filter'], $path);
+            if (is_bool($result)) {
+                return $result;
+            }
+        }
+
         if (!empty($options['except'])) {
-            if ((self::lastExcludeMatchingFromList($options['basePath'], $path, $options['except'])) !== null) {
+            if (
```

Comment:
Can be merged parent one?

Reply:
Can, but I think in current variant more understandable. What do you think?